python/doc/examples/numerical_methods/general_methods/plot_estimate_integral_iterated_quadrature.py

```diff
@@ -1,48 +1,272 @@
 """
-Estimate an integral
-====================
+Estimate a multivariate integral with IteratedQuadrature
+========================================================
 """
 # %%
-# In this example we are going to evaluate an integral of the form.
+#
+# Introduction
+# ------------
+#
+# In this example, we consider a function :math:`f: \Rset^d \mapsto \Rset`
+# and we compute the following integral:
 #
 # .. math::
-#    I_f = \int_{a}^{b}\, \int_{l_1(x_0)}^{u_1(x_0)}\, \int_{l_2(x_0, x_1)}^{u_2(x_0,x_1)}\, \int_{l_{n-1}(x_0, \dots, x_{n-2})}^{u_{n-1}(x_0, \dots, x_{n-2})}
-#    \, f(x_0, \dots, x_{n-1})\mathrm{d}{x_{n-1}}\dots\mathrm{d}{x_0}
+#    I_f = \int_{a}^{b}\, \int_{l_1(x_0)}^{u_1(x_0)}\, \int_{l_2(x_0, x_1)}^{u_2(x_0,x_1)}\, \int_{l_{d-1}(x_0, \dots, x_{d-2})}^{u_{d-1}(x_0, \dots, x_{d-2})}
+#    \, f(x_0, \dots, x_{d-1})\mathrm{d}{x_{d-1}}\dots\mathrm{d}{x_0}
 #
-# with the iterated quadrature algorithm.
+# using an iterated quadrature algorithm :class:`~openturns.IteratedQuadrature`, based on the Gauss-Kronrod
+# :class:`~openturns.GaussKronrod` 1d quadrature algorithm.
 
-# %%
 import openturns as ot
-import openturns.viewer as viewer
-from matplotlib import pylab as plt
+import openturns.viewer as otv
 import math as m
 
-ot.Log.Show(ot.Log.NONE)
+# %%
+# To get better colours for the function iso-lines.
+ot.ResourceMap.SetAsString("Contour-DefaultColorMapNorm", "rank")
+
+# %%
+# Case 1: The integrand function presents a lot of discontinuities.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# We consider the function:
+#
+# .. math::
+#    f : \begin{array}{lcl}
+#          \Rset^2 & \rightarrow & \Rset \\
+#          (x,y) & \rightarrow & 1_{{x^2+y^2 \leq 4}}(x,y)
+#          \end{array}
+#
+# the domain :math:`\set{D} =[-2,2]\times [-2,2]` and the integral:
+#
+# .. math::
+#    I = \int_{\set{D}} f(x,y)\,dxdy.
+#
+
+# %%
+# We first define the integrand :math:`f` and the domain :math:`\set{D}`.
+
+
+def kernel1(x):
+    if x[0]**2 + x[1]**2 <= 4:
+        return [1.0]
+    else:
+        return [0.0]
+
+
+integrand = ot.PythonFunction(2, 1, kernel1)
+domain = ot.Interval([-2.0, -2.0], [2.0, 2.0])
+
+# %%
+# We draw the iso-lines of the integrand function which is
+# constant equal to 1 inside the cercle and equal to 0.0 outside.
+g = integrand.draw([-3.0, -3.0], [3.0, 3.0])
+g.setTitle(r'Function $f(x,y) = 1_{\{x+2+y^2 \leq 4\}}(x,y)$')
+g.setXTitle('x')
+g.setYTitle('y')
```

Comment:
We read that we draw:

![image](https://github.com/user-attachments/assets/60510898-d7e2-433f-8177-542a231dbd91)

but there is no plot. I suggest to insert the necessary `View(g)` statement here.

Another method is to move this block of code later, where it is strictly necessary.

Reply:
yes. The graph is not very exciting but to be coherent with the treatment of the other functions, I added it.